src/main/java/com/artipie/RepoPermissions.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie;
+
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+/**
+ * Repository permissions settings.
+ * @since 0.10
+ */
+public interface RepoPermissions {
```

Comment:
@olenagerasimova I'd suggest to extract individual repository permissions class from this one, so we can avoid complex methods like `addUpdate(String, String, String)` and return type like `CompletionStage<Map<String, List<String>>>`. What I suggest is something like this:
```java
interfaces RepoPermissionsRegistry {
  List<RepoPermissions> list();
  RepoPermissions get(String name);
  ...
}
interfaces RepoPermissions {
  void add(String username, String permission);
  ...
}
```
What do you think?

Reply:
@olegmoz hm, I do not see how one more interface will allow us to avoid returning `Map<String, List<String>>` for repo permissions, as we still need map `user -> permissions list`... The only thing we will avoid here - passings parameter `repo`

Reply:
@olenagerasimova basically we replace `Map<String, List<String>>` with `RepoPermissions`. `RepoPermissions` may implement `Map<...>` or have similar methods like `Set<String> users()` instead of `Set<String> keySet()` and `Set<String> permissions(String user)` instead of `List<String> get(String value)`.

It is all about making code more understandable. It's hard to understand what is what in type like `Map<String, List<String>>`.

Reply:
@olegmoz `RepoPermissions` still will have to return users and permissions for repository (map `username -> permissions list`) as `GetPermissionSlice` have to return following json format
```json
"users" : {
      "bob": ["r","w","m"],
      "alice" : ["d","w","n", "r"]
}
```
`Map<String, List<String>>` seems to be very convenient structure to build this json. Basically, I do not mind creating the interface you are suggesting, I just do not think it's necessary in this case, we can simply rename
`CompletionStage<Map<String, List<String>>> get(String repo)` to `CompletionStage<Map<String, List<String>>> permissions(String repo)` and extend javadoc.
Furthermore, having separate methods to obtain users and permissions from settings can lead to inconsistency as they can be changed while we are building json response in `GetPermissionSlice`.

Reply:
@olenagerasimova that makes sense, thanks